packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -565,9 +575,86 @@ export function authorizationServiceBuilder(
         )
       );
     },
+
+    async createKeys({
+      clientId,
+      authData,
+      keysSeeds,
+      correlationId,
+      logger,
+    }: {
+      clientId: ClientId;
+      authData: AuthData;
+      keysSeeds: ApiKeysSeed;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<{ client: Client; showUsers: boolean }> {
+      logger.info(`Creating keys for client ${clientId}`);
+      const client = await retrieveClient(clientId, readModelService);
+      assertOrganizationIsClientConsumer(
+        unsafeBrandId(authData.organizationId),
+        client.data
+      );
+      assertKeyIsBelowThreshold(
+        clientId,
+        client.data.keys.length + keysSeeds.length
+      );
+      if (!client.data.users.includes(authData.userId)) {
+        throw userNotFound(authData.userId, authData.selfcareId);
+      }
+      await assertUserSelfcareSecurityPrivileges(
+        authData.selfcareId,
+        authData.userId,
+        authData.organizationId,
+        selfcareV2Client
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedClient: Client = client.data;
+      for (const keySeed of keysSeeds) {
+        const jwk = createJWK(decodeBase64ToPem(keySeed.key));
+        const newKey: Key = {
+          clientId,
+          name: keySeed.name,
+          createdAt: new Date(),
+          kid: calculateKid(jwk),
+          encodedPem: keySeed.key,
+          algorithm: keySeed.alg,
+          use: ApiKeyUseToKeyUse(keySeed.use),
+          userId: authData.userId,
+        };
+        const duplicateKid = await readModelService.getKeyByKid(newKey.kid);
+        if (duplicateKid) {
+          throw keyAlreadyExists(newKey.kid);
+        }
```

Comment:
This checks if any of the new keys has a duplicate in readmodel, but in this way we are not checking if the keys in the seed have a duplicate among themselves.
By the way, this shouldn't happen if the endpoint will only be used to create a single key from now on.
@galales, could we refactor it and avoid looping entirely? Without editing the api spec, we could just use the first (and only) item inside [keySeeds](https://github.com/pagopa/interop-be-monorepo/pull/575/files#diff-dbdfef4027ce7e4ab4a5697de6c66e5364588bf66e77ff0558e90d05e496dfd2R614)). What do you think?

Reply:
if in QA tests we are not using multiple values, then yes, let's consider only the first one (after checking there is at least one element).
I opened a post-migration task to refactor this API ([PIN-5033](https://pagopa.atlassian.net/browse/PIN-5033))